nixos/modules/services/mail/dovecot.nix

```
@@ -288,6 +395,28 @@ in
   options.services.dovecot2 = {
     enable = mkEnableOption "the dovecot 2.x POP3/IMAP server";
 
+    package = mkPackageOption pkgs "dovecot" {
+      default = [
+        (if versionAtLeast config.system.stateVersion "25.11" then "dovecot" else "dovecot_2_3")
+      ];
+    };
```

If this proves problematic (defaultText needing to eval `config.system.stateVersion`), you could do this:

```suggestion
    package = mkPackageOption pkgs "dovecot" { } // {
      default = if versionAtLeast config.system.stateVersion "25.11" then "dovecot" else "dovecot_2_3";
    };
```

Which will use the `description` and `defaultText` from `mkPackageOption`, but shadow the actual `default`.

IMO this is neater even if it doesn't prove necessary.